app/controllers/api/v1/subject_sets_controller.rb

```diff
@@ -5,6 +5,14 @@ class Api::V1::SubjectSetsController < Api::ApiController
   resource_actions :default
   schema_type :json_schema
 
-  allowed_params :create
-  allowed_params :update
+  protected
+
+  def build_resource_for_create(create_params)
+    super do |_, link_params|
+      if collection_id = link_params.delete("collection")
+        link_params["subjects"] = Collection.scope_for(:show, api_user)
```

Comment:
I'll fix this, it should probably just be following the normal `link_to` code path.